pkg/needacert/needacert.go

```diff
@@ -273,19 +274,23 @@ func (h *handler) generateSecret(service *corev1.Service) (*corev1.Secret, error
 	dnsNames := dnsNameSet.List()
 	secret, err := h.secretsCache.Get(service.Namespace, secretName)
```

Comment:
We are already checking here for getting the secret ? Why are we again checking in line 283 ?